python/langwatch/src/langwatch/utils/transformation.py

```
@@ -0,0 +1,218 @@
+            if hasattr(o, "model_dump"):
+                return o.model_dump(exclude_unset=True)
+            if isinstance(o, set):
+                return list(o)
+            return super().default(o)
+        except:
+            return str(o)
+
+
+def validate_safe(type_, item: dict, min_required_keys_for_pydantic_1: List[str]):
+    """Safely validate a dictionary against a type, handling both TypedDict and BaseModel."""
+    if type(item) != dict or not all(
```

Ruff (0.8.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)